### PR TITLE
py-pywcs: a new package.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    handling the FITS World Coordinate System (WCS) standard."""
+
+    homepage = "https://github.com/spacetelescope/pywcs"
+    url      = "https://github.com/spacetelescope/pywcs/archive/1.12.1.tar.gz"
+
+    version('1.12.1', sha256='efd4e0ea190e3a2521ebcde583452e126acdeac85cc8a9c78c8a96f10805b5e1')
+
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-pyfits', type=('build', 'run'))
+    depends_on('py-astropy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:20]: You're missing a lot of dependencies:
```
from setuptools import setup   

setup_requires=['d2to1>=0.2.3', 'stsci.distutils>=0.3.2'],                  

requires-python = >=2.6
requires-dist =
        astropy (>=0.3.1)
        numpy (>=1.5.1)

Comment (weijianwen) [reply var/spack/repos/builtin/packages/py-pywcs/package.py:22]: py-d2to1 and py-stsci-distutils dependencies are added.

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+class PyPywcs(PythonPackage):
+    """pywcs is a set of routines for
+    handling the FITS World Coordinate System (WCS) standard."""
+
+    homepage = "https://github.com/spacetelescope/pywcs"
+    url      = "https://github.com/spacetelescope/pywcs/archive/1.12.1.tar.gz"
+
+    version('1.12.1', sha256='efd4e0ea190e3a2521ebcde583452e126acdeac85cc8a9c78c8a96f10805b5e1')
+
+    depends_on('py-d2to1', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:18]:
```suggestion
    depends_on('py-d2to1@0.2.3:', type='build')
```

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on('py-stsci-distutils', type='build')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:19]:
```suggestion
    depends_on('py-stsci-distutils@0.3.2:', type='build')
```

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+class PyPywcs(PythonPackage):
+    """pywcs is a set of routines for
+    handling the FITS World Coordinate System (WCS) standard."""
+
+    homepage = "https://github.com/spacetelescope/pywcs"
+    url      = "https://github.com/spacetelescope/pywcs/archive/1.12.1.tar.gz"
+
+    version('1.12.1', sha256='efd4e0ea190e3a2521ebcde583452e126acdeac85cc8a9c78c8a96f10805b5e1')
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:17]: Add this:
```python
depends_on('python@2.6:', type=('build', 'run'))
depends_on('py-setuptools', type='build')
```

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on('py-d2to1', type='build')
+    depends_on('py-stsci-distutils', type='build')
+    depends_on('py-numpy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:20]:
```suggestion
    depends_on('py-numpy@1.5.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    depends_on('py-pyfits', type=('build', 'run'))
+    depends_on('py-astropy', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:22]:
```suggestion
    depends_on('py-astropy@0.3.1:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pywcs/package.py [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+    homepage = "https://github.com/spacetelescope/pywcs"
+    url      = "https://github.com/spacetelescope/pywcs/archive/1.12.1.tar.gz"
+
+    version('1.12.1', sha256='efd4e0ea190e3a2521ebcde583452e126acdeac85cc8a9c78c8a96f10805b5e1')
+
+    depends_on('py-d2to1', type='build')
+    depends_on('py-stsci-distutils', type='build')
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-pyfits', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pywcs/package.py:21]:
```suggestion
    depends_on('py-pyfits@1.4:', type=('build', 'run'))
```
For some reason, this dependency isn't listed anywhere but the README.

**Comment by weijianwen** [comment]
Version specifications are added.

**Pull request merged**